metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/syslog/Syslog5424Parser.java

```diff
@@ -61,16 +67,37 @@ public void configure(Map<String, Object> config) {
   public void init() {
   }
 
-  @Override
   @SuppressWarnings("unchecked")
+  @Override
   public List<JSONObject> parse(byte[] rawMessage) {
+    Optional<MessageParserResult<JSONObject>> resultOptional = parseOptionalResult(rawMessage);
```

Comment:
Based on the changes to the ParserBolt from https://github.com/apache/metron/pull/1184, where would this still be used?

Reply:
It wouldn't, but the interface must be implemented.  It is awkward, I'm not sure how we want to proceed with that overall.  When we added the first parseOptional and moved the bolt to it we should have deprecated the old method or something....

Reply:
I'm open to a different approach, I could throw a notimplementedexception or something.

Reply:
Just spitballing here, but maybe an option here is to create a decorator interface around the older implementation and provide defaults when someone implements the new interface? Old api doesn't change and we handle new implementations gracefully.

```java
public interface MessageParser<T> extends Configurable {
  List<T> parse(byte[] rawMessage);
  Optional<List<T>> parseOptional(byte[] parseMessage);
}
public interface MultilineMessageParser<T> extends MessageParser {
  default List<T> parse(byte[] rawMessage) {
    throw new NotImplementedException();
  }
  default Optional<MessageParserResult<T>> parseOptionalResult(byte[] parseMessage) {
    List<T> list = new ArrayList<>();
      try {
        Optional<List<T>> optionalMessages = parseOptional(parseMessage);
        optionalMessages.ifPresent(list::addAll);
      } catch (Throwable t) {
        return Optional.of(new DefaultMessageParserResult<>(t));
      }
      return Optional.of(new DefaultMessageParserResult<T>(list));
  }
}
```

**ParserBolt:**
```java
MultilineMessageParser mmp = null;
if (!(parser instanceof MultilineMessageParser)) {
  mmp = new MultilineMessageParser<JSONObject>() {

    @Override
    public void configure(Map<String, Object> config) {
      parser.configure(config);
    }

    @Override
    public void init() {
      parser.init();
    }

    @Override
    public boolean validate(JSONObject message) {
      return parser.validate(message);
    }
  };
} else {
  mmp = (MultilineMessageParser) parser;
}

Optional<MessageParserResult<JSONObject>> results = mmp.parseOptionalResult(rawMessage.getMessage());
```

Reply:
I think that is a fine idea, I had thought of a new interface, but I decided to follow the precedent of parseOptional.